eks_rolling_update.py

```diff
@@ -115,13 +115,13 @@ def update_asgs(asgs, cluster_name):
                     terminate_instance(outdated['InstanceId'])
                     if not instance_terminated(outdated['InstanceId']):
                         raise Exception('Instance is failing to terminate. Cancelling out.')
+                    detach_instance(outdated['InstanceId'], asg_name)
+                    if not instance_detached(outdated['InstanceId']):
+                        raise Exception('Instance is failing to detach from ASG. Cancelling out.')
                 except Exception as e:
                     logger.info(e)
                     raise RollingUpdateException("Rolling update on asg failed", asg_name)
 
-            # scaling cluster back down
-            logger.info("Scaling asg back down to original state")
-            scale_asg(asg_name, asg_new_desired_capacity, asg_old_desired_capacity, asg_old_max_size)
```

Comment:
@nicolasbelanger so i think we should keep this line since at the beginning of first execution the desired capacity is saved to the tags incase the rolling update fails it can start over and not keep adding more and more instances. 
If the desired count is decreased via `ShouldDecrementDesiredCapacity` then if the rolling update fails halfway through and you need to rerun you'll end up in a state where the asg desired capacity wont be back to what it originally was

Reply:
Nice catch indeed, didn't take the failure into account.  Fixing it...